src/libutil/file-system.cc

```diff
@@ -641,6 +637,13 @@ std::pair<AutoCloseFD, Path> createTempFile(const Path & prefix)
     return {std::move(fd), tmpl};
 }
 
+Path makeTempPath(const Path & root, const Path & suffix)
+{
+    // start the counter at a random value to minimize issues with preexisting temp paths
+    static std::atomic_uint_fast32_t counter(std::random_device{}());
+    return fmt("%1%%2%-%3%-%4%", root, suffix, getpid(), counter.fetch_add(1, std::memory_order_relaxed));
+}
+
```

Comment:
unfortunate there is no PRNG that is thread safe --
Would be better so code doesn't have to do the `do { } while(..)` loop.
(better trust in non-collision).

I guess here the potential collision is when _pid_ get recycled... albeit that would be incredibly rare.